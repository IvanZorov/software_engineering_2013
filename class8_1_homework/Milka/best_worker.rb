require 'csv'

best = Hash.new(0)
CSV.foreach(ARGV[0]) do |csv|
	if best[csv[2]] < csv[3].to_f
		best[csv[2]] = csv[3].to_f
	end
 end
 
 best = best.sort_by{|a,b| b.to_f}.reverse
 CSV.open("best_worker_result.csv", "w") do |csv|
 	best.each do |el|
 	csv << el
 	end
 end
