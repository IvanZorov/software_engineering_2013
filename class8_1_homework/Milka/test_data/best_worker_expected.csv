programer,60590.2
director,35755.01
cook,32601.0
taxi driver,20005.43
sport,17845.19
